rules/Php80/Rector/ClassMethod/AddParamBasedOnParentClassMethodRector.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Php80\Rector\ClassMethod;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Param;
+use PhpParser\Node\Stmt\ClassMethod;
+use PHPStan\Reflection\MethodReflection;
+use Rector\Core\PhpParser\AstResolver;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\ValueObject\PhpVersionFeature;
+use Rector\VendorLocker\ParentClassMethodTypeOverrideGuard;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://wiki.php.net/rfc/lsp_errors
+ * @see \Rector\Tests\Php80\Rector\ClassMethod\AddParamBasedOnParentClassMethodRector\AddParamBasedOnParentClassMethodRectorTest
+ */
+final class AddParamBasedOnParentClassMethodRector extends AbstractRector implements MinPhpVersionInterface
+{
+    public function __construct(
+        private readonly ParentClassMethodTypeOverrideGuard $parentClassMethodTypeOverrideGuard,
+        private readonly AstResolver $astResolver
+    ) {
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersionFeature::FATAL_ERROR_ON_INCOMPATIBLE_METHOD_SIGNATURE;
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Add missing parameter based on parent class method', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class A
+{
+    public function execute($foo)
+    {
+    }
+}
+
+class B extends A{
+    public function execute()
+    {
+    }
+}
+CODE_SAMPLE
+,
+                <<<'CODE_SAMPLE'
+class A
+{
+    public function execute($foo)
+    {
+    }
+}
+
+class B extends A{
+    public function execute($foo)
+    {
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class];
+    }
+
+    /**
+     * @param ClassMethod $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $parentMethodReflection = $this->parentClassMethodTypeOverrideGuard->getParentClassMethod($node);
+
+        if (! $parentMethodReflection instanceof MethodReflection) {
+            return null;
+        }
+
+        $parentClassMethod = $this->astResolver->resolveClassMethodFromMethodReflection($parentMethodReflection);
+        if (! $parentClassMethod instanceof ClassMethod) {
+            return null;
+        }
+
+        if ($parentClassMethod->isPrivate()) {
+            return null;
+        }
+
+        $currentClassMethodParams = $node->getParams();
+        $parentClassMethodParams  = $parentClassMethod->getParams();
+
+        if (count($currentClassMethodParams) < count($parentClassMethodParams)) {
+            $node->params = $parentClassMethodParams;
```

Comment:
replacing current classmethod with parent classmethod seems got error:

```bash
There was 1 failure:

1) Rector\Tests\Php80\Rector\ClassMethod\AddParamBasedOnParentClassMethodRector\AddParamBasedOnParentClassMethodRectorTest::test with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
assert($startPos >= 0 && $endPos >= 0) in /Users/samsonasik/www/rector-src/vendor/nikic/php-parser/lib/PhpParser/PrettyPrinterAbstract.php:535
```

need to find different solution.

Reply:
Fixed by loop parent params and re-create params with new Variable with empty attributes https://github.com/rectorphp/rector-src/pull/1431/commits/4b3bec8b5d5e6737e7bd3b7c7db688cde93b6a87